File: php/bogosort.php
<?php

function shuffleArray(&$array)
{
  for ($i = 0; $i < count($array); $i++) {
    $j = rand(0, count($array)-1);
    list($array[$i], $array[$j]) = array($array[$j], $array[$i]);
  }
}

function isSorted($array)
{
  for ($i = 1; $i < count($array); $i++) {
    if ($array[$i-1] >= $array[$i]) {
      return FALSE;
    }
  }
  return TRUE;
}

$array = array(0, 3, 2, 4, 10, 1, 7, 9);
while (!isSorted($array)) {
  shuffleArray($array);
}
print_r($array);

File: nim/bogosort.nim
import std/random

proc shuffleArray(array: var array[8, int]) =
  const length = len(array)-1
  for i in countup(0, length):
    var j = rand(length)
    swap(array[i], array[j])

proc isSorted(array: array[8, int]): bool =
  for i in countup(1, len(array)-1):
    if array[i] < array[i-1]:
      return false
  return true

randomize()
var array = [10, 2, 7, 4, 6, 1, 9, 0]
while not isSorted(array):
  shuffleArray(array)
echo array
